crates/router/src/utils/user/sample_data.rs

```diff
@@ -385,7 +405,48 @@ pub async fn generate_sample_data(
             None
         };
 
-        res.push((payment_intent, payment_attempt, refund));
+        let is_refunded = if let Some(refund) = &refund {
+            refund.refund_status == common_enums::RefundStatus::Success
+        } else {
+            false
+        };
```

Comment:
Currently all refunds of the payment have success status. We need to consider those payments for which refund has not been created. Right?

Reply:
I made this check exclusive because I thought in this way, we wouldn't need to change the dispute logic if hyperswitch decide to support various refund status in future. Otherwise it could be easy to forget to change dispute generation logic when a future issue mentions refund changes.

Reply:
I think we can remove `is_refunded`. I thought it is better to consider only payments which are not failed and for which even refund has not been initiated. For those payments it makes more sense to create a dispute.